Added unifi: Added java role, service-unifi role, service-nginx-unifi added unifi host

prisma01 opened the pull request.

belzebub40k commented:
Der Eintrag der `wireguard` Rolle in _playbooks/unifi.yml_ ist nicht nötig, dass wird bereits durch das _playbooks/services.yml_ erledigt. In _services.yml_ sind alle Basics drinnen die man braucht um einen Service-Host aufzusetzen der per Wireguard angebunden ist. Alles was host-spezifisch ist kommt in ein eigenes Playbook. Als Beispiel kannst du dir _dns.yml_ anschauen. Das heißt aber auch, dass du immer zwei Playbooks aufrufen musst, erst _services.yml_ und dann _unifi.yml_. Alternativ kannst du _unifi.yml_, wie auch _dns.yml_, in _site.yml_ importieren und dann dieses Playbook mit `--limit unifi.freifunk-mwu.de` aufrufen.

_roles/service-unifi_ ist übrigens immer noch ein kaputter Verweis auf ein anderes Git-Repo (submodule) für das es aber keinen Eintrag in _.gitmodules_ gibt.
```
fatal: No url found for submodule path 'roles/service-unifi' in .gitmodules
```

prisma01 commented:
@belzebub40k 

> roles/service-unifi ist übrigens immer noch ein kaputter Verweis auf ein anderes Git-Repo (submodule) für das es aber keinen Eintrag in .gitmodules gibt.

Hm, wo sehe ich das denn? Bei mir lokal ist das bereits aufgelöst...

belzebub40k commented:
> Hm, wo sehe ich das denn? Bei mir lokal ist das bereits aufgelöst...

In deiner Git Working Directory liegt das Repo auf das verwiesen wir vermutlich schon. Mach doch mal einen frischen Clone von dem Repo, dann solltest du sehen, dass _roles/service-unifi_ ein leeres Verzeichnis ist und `git submodule init` sollte dir diesen Fehler ausspucken.

```
fatal: No url found for submodule path 'roles/service-unifi' in .gitmodules
```

belzebub40k commented:
Jetzt sieht es gut aus. Ich würde noch die _service-nginx-unifi_ Rolle in die _service-unifi_ Rolle integrieren. Bei anderen Service-Rollen die vHosts anlegen, z.B. _service-grafana_, haben wir das auch nicht getrennt.

Die _service-nginx-unms_ Rolle würde ich eventuell komplett entfernen solange wir UNMS nicht komplett mit Ansible managen.

prisma01 commented:
> Jetzt sieht es gut aus. Ich würde noch die service-nginx-unifi Rolle in die service-unifi Rolle integrieren. Bei anderen Service-Rollen die vHosts anlegen, z.B. service-grafana, haben wir das auch nicht getrennt.

Ja, das ist sinnvoll. Done.

> Die service-nginx-unms Rolle würde ich eventuell komplett entfernen solange wir UNMS nicht komplett mit Ansible managen.

Das heisst nicht gemangte vhosts im Nginx? Das finde ich uncool. Ich habe die Readme der Rolle nochmal angepasst, damit klar ist, dass das kein UNMS installiert.

belzebub40k commented on inventory/host_vars/unifi.freifunk-mwu.de:
```diff
@@ -0,0 +1,4 @@
+---
+server_type: "service"
+
+magic: 260
```
Die `magic` muss noch auf einen Wert < 255 geändert werden. Momentan erzeugen einige Rollen unbrauchbare Ausgaben. z.B. spuckt die Rolle für das Loopback Interface `False` als IPv4 Adresse aus.

prisma01 replied:
Hm, würde ich ja gerne machen. Aber eigentlich ist das 10.X.0.0/24 ja für gateways reserviert. Oder ignorieren wir das mal? Wäre es nicht sinnvoller die Rollen zu fixen?

belzebub40k replied:
Die anderen Server (suesskartoffel, linse und kichererbse) sind mittlerweile auch alle in 10.X.0.0/24. Wir werden vermutlich nie 253 Gateways haben ;-) Außerdem sind die neuen Multidomain Netze (/20) kleiner als die zwei Legacy Netze (/18) so das es generell sinnvoll ist sparsamer mit dem Adressbereich umzugehen.

prisma01 replied:
Ok.

kokel replied:
Die neuen Netzbereiche gibts hier: https://md.freifunk-mwu.de/domain-split#Netzbereiche1
Das Netz 10.87.255.0/24 ist für Loopbacks vorgesehen. Im Normalfall hat ein Dienste-Server nur eine Loopback in diesem Netz, auf die man aus den Domain geroutet zugreifen kann. Wenn ein Dienst ein Layer2-Bein in einer oder mehreren Domains benötigt, sieht das natürlich anders aus.

belzebub40k commented:
Ich hab dann keine Einwände mehr aber am besten squash-en wir die Commits noch vor dem Merge.

prisma01 committed:
Added Unfi Server to ansible.

Squashed commit of the following:

commit 3ba76f9e3c02e4a343c0af62ce0327dee6f2f109
Author: Julian Labus <julian@labus-online.de>
Date:   Thu Aug 22 13:18:05 2019 +0200

    make unifi.yml executable

commit 36002be43a7c1af76660682469dc7efcc06fac5f
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 13:10:32 2019 +0200

    Changed magic < 255

commit c26bbac23620455653d8e21bf512c48493362d43
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 12:05:42 2019 +0200

    Ergänzug um Installation von UNMS

commit 220ba24e471f8139d775eec1d4b3da46e8befdf7
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 12:01:38 2019 +0200

    moved service-nginx-unifi into service-unifi

commit 570f5673c47b72e959e6da8940a9cbcbf1e496bf
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 11:40:27 2019 +0200

    remove lean_delivery.java from unifi playbook. already in dependencies of service-unifi

commit 4775cff44d53e07b720e8760f0e4ece1460e0717
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 11:38:44 2019 +0200

    adjusted readme, remove galaxy info, removed unit test

commit 75711495fcb40f618cd65029ceda6b3cfed7e581
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 11:32:59 2019 +0200

    fix type in unifi inventory

commit 513b4f6b8f3d1426e58a19ece6514e467916d680
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 11:32:44 2019 +0200

    move service-unifi back to correct location

commit ea32af8275fd7e6c34281e02eeb2323cbea55f64
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 11:31:42 2019 +0200

    move service-unifi to temp directory

commit 87698c8d84a66cbf844e923c6dd41de87cba7759
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 02:09:17 2019 +0200

    Move host definition to inentory

commit b469c3f6307dbed9f40310c590abc0c672487973
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Thu Aug 22 01:52:19 2019 +0200

    Add unifi.yml to site.yml, remove wireguard role from unifi.yml

commit 77fa3aa0eb04b01a7b10dd66cede41b09e0aa5c0
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Wed Aug 14 21:17:49 2019 +0200

    Add unifi to wireguard

commit c63b850ddd470aa43d8ec25663f086180157673d
Author: Julian Labus <julian@labus-online.de>
Date:   Tue Aug 13 08:35:53 2019 +0200

    Add wireguard_networks for host unifi

commit 21341ad359fcfe9a9e4735ca50750f1bd5ea2ab6
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Mon Aug 12 22:49:17 2019 +0200

    Add unms nginx vhost

commit 0c0dff99b6ddbf8c5a354cce040b649c43948d64
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Mon Aug 12 21:44:30 2019 +0200

    Add java role to gitignore

commit 4fa292b5e5528b732e71dfd204ce53878cc3ee7e
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Mon Aug 12 21:43:50 2019 +0200

    minor changes

commit 25ab40e85e8b2fd6eb4a770e4c26a82894db3622
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Mon Aug 12 21:17:39 2019 +0200

    move java repo to requirements.yml, replace duplicated serice-nginx with service-nginx + dependency

commit d9aa004149da7ac3893b698d4dcb329236a521fe
Author: Sebastian Schmachtel <prisma_freifunk@oimel.net>
Date:   Mon Aug 12 00:52:24 2019 +0200

    Added unif host: Added java role, service-unifi role, service-nginx-unifi added unifi host

prisma01 commented:
Gut, ist gesquashed.
